File: java/src/test/java/problems101_200/SingleNumberTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SingleNumberTest extends SampledTest {
    @Test
    public void test() {
        for (SingleNumber solution : SingleNumber.values()) {
            for (SingleNumberSample sample : getSamples(SingleNumberSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.singleNumber(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/SingleNumberTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class SingleNumberTest : SampledTest() {
    @Test
    fun test() =
        SingleNumber.entries.forEach { solution ->
            getSamples(Array<SingleNumberSample>::class.java).forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.singleNumber(input.nums))
                    .isEqualTo(output)
            }
        }
}
